src/spec-node/dockerCompose.ts

```diff
@@ -499,6 +508,20 @@ async function generateFeaturesComposeOverrideContent(
 	const userCommand = overrideCommand ? [] : composeCommand /* $ already escaped. */
 		|| (composeEntrypoint ? [/* Ignore image CMD per docker-compose.yml spec. */] : ((await imageDetails()).Config.Cmd || []).map(c => c.replace(/\$/g, '$$$$'))); // $ > $$ to escape docker-compose.yml's interpolation.
 
+
+	const hasGpuRequirement = config.hostRequirements?.gpu;
+	const addGpuCapability = hasGpuRequirement && await checkDockerSupportForGPU(params);
+	if (hasGpuRequirement && hasGpuRequirement !== 'optional' && !addGpuCapability) {
+		throw Error('Unable to detect GPU yet a GPU was required - consider marking it as "optional"');
```

Comment:
We realized that we don't enforce other requirements. Having a warning would still be valuable, could you write this message to `output.write` with `LogLevel.Warning` instead of throwing an error?